packages/event-store/src/EventStore/FileSystemAdapter/JSONLinesFileSystemAdapter.js

```diff
@@ -0,0 +1,74 @@
+/* @flow */
+
+import fs from 'fs';
+import { Readable, Writable, PassThrough } from 'stream';
+import jsonlines from 'jsonlines';
+import { stringify } from 'querystring';
+import { FileSystemAdapter } from './../FileSystemAdapter';
+import { transformify } from './../../utils/stream';
+
+const createFilterByStreamNameStream = (streamName: string) => {
+  return transformify(
+    (data, encoding, cb) =>
+      data.stream_name === streamName ? cb(null, data) : cb(null, ''),
+  );
+};
+
+const createExtractPayloadStream = () => {
+  return transformify((data, encoding, cb) => cb(null, data.event));
+};
+
+const createEncapsulatePayloadStream = (streamName: string) => {
```

Comment:
2 verbs in a single action, is it a good idea?

Reply:
This is not perfect. I do this keep concistancy with the node `fs.create[Read|Write]Stream` method. I will simplify this